tests/util.spec.coffee

```diff
@@ -0,0 +1,119 @@
+m = require('mochainon')
+
+{ mergePineOptions } = require('../lib/util')
+
+describe 'Pine option merging', ->
+
+	it 'uses the defaults only, if no extra options are provided', ->
+		defaults = filter: id: 1
+		result = mergePineOptions(defaults, undefined)
+		m.chai.expect(result).to.deep.equal(defaults)
+
+	it "uses extra options directly if they don't conflict with defaults", ->
+		extras =
+			filter: id: 1
+			select: [ 'id' ]
+			expand:
+				device:
+					$select: ['id']
+					$expand: ['application', 'user']
+			top: 1
+			skip: 1
+		result = mergePineOptions({}, extras)
+		m.chai.expect(result).to.deep.equal(extras)
+
+	it 'overrides select, top, skip and orderby options', ->
+		result = mergePineOptions
+			top: 1
+			skip: 2
+			select: [ 'id' ]
+			orderby: 'app_name asc'
+		,
+			top: 3
+			skip: 4
+			select: [ 'app_name' ]
+			orderby: 'id asc'
+
+		m.chai.expect(result).to.deep.equal
+			top: 3
+			skip: 4
+			select: [ 'app_name' ]
+			orderby: 'id asc'
+
+	it 'combines filter options with $and', ->
+		result = mergePineOptions
+			filter: id: 1
+		,
+			filter: name: 'MyApp'
+
+		m.chai.expect(result).to.deep.equal
+			filter:
+				$and: [
+					id: 1
+				,
+					name: 'MyApp'
+				]
+
+	it 'combines expand options for separate single relationships', ->
+		result = mergePineOptions
+			expand: 'device'
+		,
+			expand: 'application'
+
+		m.chai.expect(result).to.deep.equal
+			expand:
+				device: {}
```

Comment:
Is it actually valid to pass an empty `expand` object to pinejs-client (I honestly don't remember - it's not something you'd normally do by hand, but makes sense in this auto-generated form)